uportal-war/src/main/java/org/jasig/portal/rendering/StylesheetAttributeSource.java

```diff
@@ -133,4 +150,35 @@ public IStylesheetDescriptor getStylesheetDescriptor(HttpServletRequest request)
     }
     
     public abstract PreferencesScope getStylesheetPreferencesScope(HttpServletRequest request);
+
+    protected abstract Logger getLogger();
+
+    protected boolean shouldDoSpelEvaluationForAttributeValue(final String attributeValue) {
+        return this.portalSpELService != null && attributeValue != null && attributeValue.startsWith("${") && attributeValue.endsWith("}");
+    }
+
+    protected String getValueForSpelEvaluation(final String attributeValue) {
+        if (attributeValue != null && attributeValue.startsWith("${") && attributeValue.endsWith("}")) {
+            return attributeValue.substring(2, attributeValue.length()-1);
+        } else {
+            return attributeValue;
+        }
+    }
+
+    protected String doSpelEvaluationForAttributeValue(final WebRequest request, final String attributeValue) {
+        String result;
+        try {
+            final String valueForEvaluation = this.getValueForSpelEvaluation(attributeValue);
+            final Expression expression = this.portalSpELService.parseExpression(valueForEvaluation);
+            result = this.portalSpELService.getValue(expression, request, String.class);
```

Comment:
Minor but the above 3 lines could be simplified to
```
return this.portalSpELService.getValue(getValueForSpelEvaluation(attributeValue), request, String.class);
```
It would also allow you to avoid the try/catch block.